conda/connection.py

```diff
@@ -107,8 +108,11 @@ def get_binstar_token(url):
             return None
 
     @staticmethod
-    def is_binstar_url(url):
+    def is_binstar_url_needing_token(url):
         urlparts = urlparse(url)
+        token, path = split_token(urlparts.path)
```

Comment:
Can split_token() ever fail?  Is there a situation where it doesn't fail, but returns an invalid/empty token?